lib/index.js

```diff
@@ -221,6 +221,9 @@ class NightwatchClient extends EventEmitter {
 
   setReporter(reporter) {
     this.__reporter = reporter;
+    if (reporter && typeof reporter.setClient === 'function') {
```

Comment:
It's better if we create an separate, independent screenshots service, which will be located in the transport folder. This service should have access to the `nightwatchInstance` so it knows the currently running test but it shouldn't queue the screenshot action, but rather call the `getScreenshot` action directly.

Reply:
@beatfactor, I have create a screenshots services at `transport/services`.